broadcast_admin.php

```diff
@@ -17,19 +17,28 @@
 // URL parameters
 $action = $_REQUEST['action'];
 $numbers = $_POST['numbers'];
+$tags = $_POST['tags'];
 $send_welcome = ($_POST['send_welcome'] == 'on');
+$id = $_REQUEST['id'];
+$tag = $_REQUEST['tag'];
 
 // *** ACTIONS ***
 
 // import?
 if ($action == 'import') {
-	importNumbers($numbers, $error, $success, $send_welcome);
+	importNumbers($numbers, $tags, $error, $success, $send_welcome);
 // remove?
 } else if ($action == 'remove') {
 	removeNumbers($numbers, $error, $success);
+// remove a tag?
+} else if ($action == 'removetag') {
+	removeTag($id, $tag, $error);
+	$action = 'list';
+}
+
 // list?
-} else if ($action == 'list') {
-	loadNumbers($numbers_active, $numbers_disabled, $error);
+if (substr($action, 0, 4) == 'list') {
+	loadNumbers($numbers_active, $numbers_disabled, $numbers_by_tag, $error);
```

Comment:
Is it possible to pass by reference an uninitialized variable in PHP, thus causing it to be initialized by the function to which it is passed? Maybe that's what I'm missing in reviewing this code. I don't see $numbers_by_tag initialized above this.

Reply:
Yes, see above.